core/trino-main/src/main/java/io/trino/execution/PartitionedSplitsInfo.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+
+public final class PartitionedSplitsInfo
+{
+    private static final PartitionedSplitsInfo NO_SPLITS_INFO = new PartitionedSplitsInfo(0, 0);
+
+    private final int count;
+    private final long weightSum;
+
+    private PartitionedSplitsInfo(int splitCount, long splitsWeightSum)
+    {
+        this.count = splitCount;
+        this.weightSum = splitsWeightSum;
+    }
+
+    public int getCount()
+    {
+        return count;
+    }
+
+    public long getWeightSum()
+    {
+        return weightSum;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return (count * 31) + Long.hashCode(weightSum);
```

Comment:
Shouldn't _really_ matter here, especially since `Integer.hashCode` is implemented as:
```
public static int hashCode(int value) {
    return value;
}
```

Reply:
Haha, totally forgot that's the way Java implements it :-)